ui/dashboards/src/components/Panel/PanelActions.tsx

```diff
@@ -143,59 +180,73 @@ export const PanelActions: React.FC<PanelActionsProps> = ({
-      {editActions && (
-        <>
-          <Box
-            sx={combineSx((theme) => ({
-              display: 'block',
-              [theme.containerQueries(HEADER_ACTIONS_CONTAINER_NAME).down(HEADER_ACTIONS_MIN_WIDTH)]: {
-                display: 'none',
-              },
-            }))}
-          >
-            {editActions}
-          </Box>
-          <Box
-            sx={combineSx((theme) => ({
-              display: 'block',
-              [theme.containerQueries(HEADER_ACTIONS_CONTAINER_NAME).up(HEADER_ACTIONS_MIN_WIDTH)]: { display: 'none' },
-            }))}
-          >
-            <ShowAction title={title}>{editActions}</ShowAction>
-          </Box>
-        </>
-      )}
-      {editActions && !readHandlers?.isPanelViewed && (
+  const moveAction = useMemo(() => {
+    if (editActions && !readHandlers?.isPanelViewed) {
+      return (
         <InfoTooltip description={TOOLTIP_TEXT.movePanel}>
           <HeaderIconButton aria-label={ARIA_LABEL_TEXT.movePanel(title)} size="small">
             <DragIcon className="drag-handle" sx={{ cursor: 'grab' }} fontSize="inherit" />
           </HeaderIconButton>
         </InfoTooltip>
-      )}
-    </HeaderActionWrapper>
+      );
+    }
+    return undefined;
+  }, [editActions, readHandlers, title]);
+
+  const divider = <Box sx={{ flexGrow: 1 }}></Box>;
+
+  return (
+    <>
+      {/* small panel width: move all icons except move/grab to overflow menu */}
+      <ConditionalBox
+        sx={(theme) => ({
+          [theme.containerQueries(HEADER_ACTIONS_CONTAINER_NAME).between(0, HEADER_SMALL_WIDTH)]: { display: 'flex' },
+        })}
+      >
+        {divider}
+        <OverflowMenu title={title}>
+          {descriptionAction} {linksAction} {queryStateIndicator} {extraActions} {readActions} {editActions}
+        </OverflowMenu>
+        {moveAction}
+      </ConditionalBox>
+
+      {/* medium panel width: move edit icons to overflow menu */}
+      <ConditionalBox
+        sx={(theme) => ({
+          [theme.containerQueries(HEADER_ACTIONS_CONTAINER_NAME).between(HEADER_SMALL_WIDTH, HEADER_MEDIUM_WIDTH)]: {
+            display: 'flex',
+          },
+        })}
+      >
+        {descriptionAction} {linksAction} {divider} {queryStateIndicator} {extraActions} {readActions}
+        <OverflowMenu title={title}>{editActions}</OverflowMenu>
+        {moveAction}
+      </ConditionalBox>
+
+      {/* large panel width: show all icons in panel header */}
+      <ConditionalBox
+        sx={(theme) => ({
+          // flip the logic here; if the browser (or jsdom) does not support container queries, always show all icons
+          display: 'flex',
+          [theme.containerQueries(HEADER_ACTIONS_CONTAINER_NAME).down(HEADER_MEDIUM_WIDTH)]: { display: 'none' },
+        })}
+      >
+        {descriptionAction} {linksAction} {divider} {queryStateIndicator} {extraActions} {readActions} {editActions}{' '}
```

Comment:
Can be re removed, no? I guess it's prettier that added it 🤔 
```suggestion
        {descriptionAction} {linksAction} {divider} {queryStateIndicator} {extraActions} {readActions} {editActions}
```

Reply:
Yep, prettier added this one. I'll remove it.